feature-libs/cart/base/core/http-interceptors/handlers/bad-cost-center-request.handler.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable } from "@angular/core";
+import { HttpErrorResponse, HttpRequest } from "@angular/common/http";
+import { isEntityValidationError} from "../../utils/utils";
+import {
+  ErrorModel,
+  GlobalMessageType,
+  HttpErrorHandler,
+  HttpResponseStatus,
+  Priority,
+} from '@spartacus/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class BadCostCenterRequestHandler extends HttpErrorHandler {
+  responseStatus = HttpResponseStatus.BAD_REQUEST;
+
+  getPriority(): Priority {
+    return Priority.NORMAL;
+  }
+
+  hasMatch(errorResponse: HttpErrorResponse): boolean {
+    return (
+      super.hasMatch(errorResponse) &&
+      this.getErrors(errorResponse).some(isEntityValidationError) &&
+      this.isCostCenterRequest(errorResponse)
+    );
+  }
+
+  handleError(_request: HttpRequest<any>, response: HttpErrorResponse): void {
+    this.getErrors(response)
+      .filter((e) => isEntityValidationError(e))
+      .forEach(() => {
+        this.globalMessageService.add(
+          { key: 'httpHandlers.invalidCostCenter' },
+          GlobalMessageType.MSG_TYPE_ERROR
+        );
+      });
+  }
+
+  protected getErrors(response: HttpErrorResponse): ErrorModel[] {
+    return (response.error?.errors || []).filter(
+      (error: any) => error.type !== 'JaloObjectNoLongerValidError'
+    );
+  }
+
+  private isCostCenterRequest(errorResponse: HttpErrorResponse): boolean {
+    if (errorResponse?.url) {
+      const url = new URL(errorResponse.url);
+      return url.pathname.endsWith('costcenter') && new URLSearchParams(url.search).has('costCenterId');
```

Comment:
The url is configurable.  I wonder if there is another way to narrow down to which response the error is applicable.

Reply:
There's no other way I can think of to identify the response as a cost center request. The only response we get is
`{
   "errors" : [ {
      "message" : "The application has encountered an error",
      "type" : "EntityValidationError"
   } ]
}`

Reply:
This may be quite difficult to do, but you could use the coded value in `defaultOccCostCentersConfig` in the file `default-occ-cost-centers-config.ts`.

Reply:
Based on discussion with Patrick we decided to keep it as is.